apis/core/v1beta1/openstackcontrolplane_types.go

```diff
@@ -485,6 +489,21 @@ type OctaviaSection struct {
 	Template octaviav1.OctaviaSpec `json:"template,omitempty"`
 }
 
+// RedisSection defines the desired state of the Redis service
+type RedisSection struct {
+	// +kubebuilder:validation:Optional
+	// Enabled - Whether the Redis service should be deployed and managed
+	// +kubebuilder:default=false
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:booleanSwitch"}
+	Enabled bool `json:"enabled"`
+
+	// +kubebuilder:validation:Optional
+	//+operator-sdk:csv:customresourcedefinitions:type=spec
+	// Templates - Overrides to use when creating the Redis Resources
+	Templates map[string]redisv1.RedisSpec `json:"templates,omitempty"`
```

Comment:
If you need any defulting or validating webhook calls on the RedisSpec then you need to do the calls in openstackcontrolplane_webhook.go. ( I at least see a container image defaulting webhook on it)

Reply:
I think the latest update should have this covered.